### DISCUSSION TITLE
Collect time series data

### DISCUSSION THREAD

**Discussion by narcolis** [DanWBR/dwsim#609] [Q&A]
Hi @DanWBR,
Thank you for your great contribution! 
I want to collect time series data for Reinforcement training, which requires that the output of current time will be the input for the next step. However, I'm not so clear with  chemical process, so I want to utilize the completed samples. Could you give me some suggestions on which type of simulator may fits my needs? Thank you so much!

**Comment by Spogis** [comment]
Dear @narcolis , how are you getting on?

I have been working with Daniel on various cases, mainly in the area of Machine Learning applied within DWSIM. 
Could you explain in more detail your needs so we can see how we can help you? At first, if you are working with Time Series, it would be necessary to use DWSIM in dynamic mode, which requires a bit more practice in simulators.
I appreciate your attention in advance.
Sincerely,
Nicolas Spogis

**Comment by Spogis** [comment]
@narcolis , this is a interesting paper: 
Transfer learning for process design with reinforcement learning. 

Prof. Artur M. Schweidtmann ( a.schweidtmann@tudelft.nl) 
https://arxiv.org/ftp/arxiv/papers/2302/2302.03375.pdf

**Reply by narcolis** [reply]
Hi @Spogis , 
Thank you for your kindly reply! 
I have read the paper **_Transfer learning for process design with RL_**, it used DWSIM to calculate the reward for each steps in RL training, which is slightly different from my needs. 

My purpose is to collect some trajectory data  in the field of chemical engineering. Thus, I need to define the state, action and reward in a simulator . For example, the **states** can include information such as temperature, pressure, operational states of equipment, etc. **Actions** can involve adjusting equipment operations， tuning reactor parameters, and  **rewards** may be the goals of the chemical process, such as maximizing product yields, or minimizing energy consumption.

I think some simulators may not suitable to collect the trajectories, because the trajectories must be sequential and include state, action and reward at each step. Could you give me some suggestions that which kinds of simulator may be useful for me? 
I appreciate your help a lot!

Sincerely,
narcolis

**Reply by Spogis** [reply]
Dear @narcolis,

You can perform this type of simulation within DWSIM without any issues, but it involves a significant prior understanding of the software, knowledge of Python, and primarily the DWSIM API.

You can utilize the Automation tools provided by DWSIM, as described here: https://dwsim.org/wiki/index.php?title=Automation

Another interesting point for you to study would be the Dynamic Simulation Tutorial with DWSIM and Python: https://dwsim.org/wiki/index.php?title=Dynamic_Simulation_Tutorial_with_DWSIM_and_Python,_Part_1:_Concepts_and_Steady-State_Model

I believe with this information, you will be able to carry out your project successfully.

Best regards,

Nicolas
